# Global YouTube Statistics 2023

Welcome to the captivating realm of YouTube stardom, where this meticulously curated dataset unveils the statistics of the most subscribed YouTube channels. A collection of YouTube giants, this dataset offers a perfect avenue to analyze and gain valuable insights from the luminaries of the platform. With comprehensve details on top creators' subscriber counts, video views, upload frequency, country of origin, earnings, and more, this treasure trove of infromation is a must-explore for aspiring content creators, data enthusiasts, and anyone intrigued by the ever-evolving online content landscape. Immerse yourself in the world of Youtube success and unlock a wealth of knowledge with this extraordinary dataset.

# Project:
Today we are going to use the available data to assist us with our new idea of starting a YouTube channel! Utilizing our knowledge of analyzing, filtering, and visualizing data, we are going to try to figure out if there is a specific idea or category of videos that we should focus on for our channel. Things that we should take into consideration are: fan-base sizes, channel growth-rates, moneys earned, outreach/collaboration possibilities, and anything else we may stumble accross during our research phase that might affect our decision of what kinds of videos to make.

Things we are going to explore:
- YouTube categories
- Moneys earned
- Related Channels
- Location

## Outline:
**Category**
1) What categories are most prevalent in the top rankings?
    - Is too much saturation a good or bad thing?
3) What cateogry has the highest overall amount of subscribers?

**Moneys Earned**
1) Is there a relationship between more moneys based on categories?
2) Is there a relationship between more moneys based on location?

**Help/Outreach**
1) What are some channels in each category that we could potentially reach out to for help/networking?
2) We may want to focus on lower sub-count/rank channels because they may be more likely to collab with us.

**Location**
1) Whats the spread of categories based on location?

**Trending**
1) What categories have been recently trending?
    - Consider views and subscribers.

## EDA

Before starting to analyze the data to answer the questions above, let's first get some exploratory summary data and see if we need to consider any cleaning of our data.

Information from Kaggle about the columns within the dataset:
Columns:
- rank: Position of the YouTube channel based on the number of subscribers
- Youtuber: Name of the YouTube channel
- subscribers: Number of subscribers to the channel
- video views: Total views across all videos on the channel
- category: Category or niche of the channel
- Title: Title of the YouTube channel
- uploads: Total number of videos uploaded on the channel
- Country: Country where the YouTube channel originates
- Abbreviation: Abbreviation of the country
- channel_type: Type of the YouTube channel (e.g., individual, brand)
- video_views_rank: Ranking of the channel based on total video views
- country_rank: Ranking of the channel based on the number of subscribers wi thin its country
- channel_type_rank: Ranking of the channel based on its type (individual or brand)
- video_views_for_the_last_30_days: Total video views in the last 30 days
- lowest_monthly_earnings: Lowest estimated monthly earnings from the channel
- highest_monthly_earnings: Highest estimated monthly earnings from the channel
- lowest_yearly_earnings: Lowest estimated yearly earnings from the channel
- highest_yearly_earnings: Highest estimated yearly earnings from the channel
- subscribers_for_last_30_days: Number of new subscribers gained in the last 30 days
- created_year: Year when the YouTube channel was created
- created_month: Month when the YouTube channel was created
- created_date: Exact date of the YouTube channel's creation
- Gross tertiary education enrollment (%): Percentage of the population enrolled in tertiary education in the country
- Population: Total population of the country
- Unemployment rate: Unemployment rate in the country
- Urban_population: Percentage of the population living in urban areas
- Latitude: Latitude coordinate of the country's location
- Longitude: Longitude coordinate of the country's location

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

data = pd.read_csv('Global_Youtube_Statistics.csv', encoding='latin1')

In [2]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 995 entries, 0 to 994
Data columns (total 28 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   rank                                     995 non-null    int64  
 1   Youtuber                                 995 non-null    object 
 2   subscribers                              995 non-null    int64  
 3   video views                              995 non-null    float64
 4   category                                 949 non-null    object 
 5   Title                                    995 non-null    object 
 6   uploads                                  995 non-null    int64  
 7   Country                                  873 non-null    object 
 8   Abbreviation                             873 non-null    object 
 9   channel_type                             965 non-null    object 
 10  video_views_rank                         994 non-n

**EDA:**
- Number of rows
- Number of Categories/list of
- Number of Countries/list of
- Summary of dtypes
- Statistical summaries of numerical datatypes
- Short view of overall dataset

In [3]:
print(len(data))

995


In [4]:
print(data.category.nunique())
print(data.category.unique())

18
['Music' 'Film & Animation' 'Entertainment' 'Education' 'Shows' nan
 'People & Blogs' 'Gaming' 'Sports' 'Howto & Style' 'News & Politics'
 'Comedy' 'Trailers' 'Nonprofits & Activism' 'Science & Technology'
 'Movies' 'Pets & Animals' 'Autos & Vehicles' 'Travel & Events']


In [5]:
print(data.Country.nunique())
print(data.Country.unique())

49
['India' 'United States' nan 'Japan' 'Russia' 'South Korea'
 'United Kingdom' 'Canada' 'Brazil' 'Argentina' 'Chile' 'Cuba'
 'El Salvador' 'Pakistan' 'Philippines' 'Thailand' 'Colombia' 'Barbados'
 'Mexico' 'United Arab Emirates' 'Spain' 'Saudi Arabia' 'Indonesia'
 'Turkey' 'Venezuela' 'Kuwait' 'Jordan' 'Netherlands' 'Singapore'
 'Australia' 'Italy' 'Germany' 'France' 'Sweden' 'Afghanistan' 'Ukraine'
 'Latvia' 'Switzerland' 'Vietnam' 'Malaysia' 'China' 'Iraq' 'Egypt'
 'Andorra' 'Ecuador' 'Morocco' 'Peru' 'Bangladesh' 'Finland' 'Samoa']


In [6]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 995 entries, 0 to 994
Data columns (total 28 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   rank                                     995 non-null    int64  
 1   Youtuber                                 995 non-null    object 
 2   subscribers                              995 non-null    int64  
 3   video views                              995 non-null    float64
 4   category                                 949 non-null    object 
 5   Title                                    995 non-null    object 
 6   uploads                                  995 non-null    int64  
 7   Country                                  873 non-null    object 
 8   Abbreviation                             873 non-null    object 
 9   channel_type                             965 non-null    object 
 10  video_views_rank                         994 non-n

In [7]:
print(data.describe())

            rank   subscribers   video views        uploads  video_views_rank  \
count  995.00000  9.950000e+02  9.950000e+02     995.000000      9.940000e+02   
mean   498.00000  2.298241e+07  1.103954e+10    9187.125628      5.542489e+05   
std    287.37606  1.752611e+07  1.411084e+10   34151.352254      1.362782e+06   
min      1.00000  1.230000e+07  0.000000e+00       0.000000      1.000000e+00   
25%    249.50000  1.450000e+07  4.288145e+09     194.500000      3.230000e+02   
50%    498.00000  1.770000e+07  7.760820e+09     729.000000      9.155000e+02   
75%    746.50000  2.460000e+07  1.355470e+10    2667.500000      3.584500e+03   
max    995.00000  2.450000e+08  2.280000e+11  301308.000000      4.057944e+06   

       country_rank  channel_type_rank  video_views_for_the_last_30_days  \
count    879.000000         962.000000                      9.390000e+02   
mean     386.053470         745.719335                      1.756103e+08   
std     1232.244746        1944.386561    

In [8]:
print(data.head())

   rank                    Youtuber  subscribers   video views  \
0     1                    T-Series    245000000  2.280000e+11   
1     2              YouTube Movies    170000000  0.000000e+00   
2     3                     MrBeast    166000000  2.836884e+10   
3     4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4     5                   SET India    159000000  1.480000e+11   

           category                       Title  uploads        Country  \
0             Music                    T-Series    20082          India   
1  Film & Animation               youtubemovies        1  United States   
2     Entertainment                     MrBeast      741  United States   
3         Education  Cocomelon - Nursery Rhymes      966  United States   
4             Shows                   SET India   116536          India   

  Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
0           IN          Music  ...                     2000000.0   
1           US  

**EDA Notes:**
- Number of rows
    - 995 (YouTubers)
- Number of Categories/list of
    - 18
- Number of Countries/list of
    - 49
- Summary of dtypes
    - All variables look to be categorized under the correct datatypes.
    - 'subscribes_for_last_30_days' only hs 658 non-null values
- Statistical summaries of numerical datatypes
   - uploads and video_views both have minimum values of 0, maybe some incorrect entries?
   - upload_year has a minimum value of 1970 but YouTube wasn't created until 2005. Also, the internet wasn't a thing in 1970...
- Short view of overall dataset
    - Will find out more upon further exploration.

## Data Analysis:

**Category (C)**
1) What categories are most prevalent in the top rankings?
    - Is too much saturation a good or bad thing?
2) What cateogry has the highest overall amount of subscribers?

In [61]:
# C1 - What categories are most prevalent in the top rankings?
print(data.category.value_counts().reset_index())

                 category  count
0           Entertainment    241
1                   Music    202
2          People & Blogs    132
3                  Gaming     94
4                  Comedy     69
5        Film & Animation     46
6               Education     45
7           Howto & Style     40
8         News & Politics     26
9    Science & Technology     17
10                  Shows     13
11                 Sports     11
12         Pets & Animals      4
13               Trailers      2
14  Nonprofits & Activism      2
15                 Movies      2
16       Autos & Vehicles      2
17        Travel & Events      1


In [60]:
# C2: What category has the highest overall amount of subscribers?
print(data[['category', 'subscribers']].groupby('category').sum('subscribers').sort_values('subscribers', ascending=False).reset_index())

                 category  subscribers
0                   Music   5195000000
1           Entertainment   5158200000
2          People & Blogs   2779400000
3                  Gaming   1960100000
4                  Comedy   1388500000
5        Film & Animation   1314900000
6               Education   1194400000
7           Howto & Style    775600000
8                   Shows    541000000
9         News & Politics    536400000
10   Science & Technology    316500000
11                 Sports    298200000
12               Trailers     78000000
13         Pets & Animals     72400000
14  Nonprofits & Activism     55500000
15                 Movies     51300000
16       Autos & Vehicles     35700000
17        Travel & Events     12500000


**Moneys Earned (M)**
1) Is there a relationship between more moneys based on categories?
2) Is there a relationship between more moneys based on location?

In [59]:
# M1: Is there a relationship between more moneys based on categories?
# Columns dealing with moneys:
# lowest_monthly_earnings, highest_monthly_earnings, lowest_yearly_earnings, highest_yearly_earnings

# PART 1 -- highest_monthly_earnings
print(data[['category', 'highest_monthly_earnings']].groupby('category').mean('highest_monthly_earnings').sort_values('highest_monthly_earnings', ascending=False).reset_index())

                 category  highest_monthly_earnings
0                   Shows              2.037662e+06
1        Autos & Vehicles              1.086350e+06
2                  Sports              8.069364e+05
3          Pets & Animals              7.944322e+05
4        Film & Animation              7.327018e+05
5               Education              6.951778e+05
6                  Comedy              6.683122e+05
7         News & Politics              6.426320e+05
8           Entertainment              6.226426e+05
9                   Music              5.522996e+05
10         People & Blogs              5.468421e+05
11                 Movies              4.547000e+05
12  Nonprofits & Activism              3.904000e+05
13               Trailers              3.619000e+05
14                 Gaming              2.719054e+05
15   Science & Technology              2.020432e+05
16          Howto & Style              1.927991e+05
17        Travel & Events              1.240000e+05


In [12]:
# M1: Is there a relationship between more moneys based on categories?
# Columns dealing with moneys:
# lowest_monthly_earnings, highest_monthly_earnings, lowest_yearly_earnings, highest_yearly_earnings

# PART 2 -- lowest_monthly_earnings
print(data[['category', 'lowest_monthly_earnings']].groupby('category').mean('lowest_monthly_earnings').sort_values('lowest_monthly_earnings', ascending=False))

# # create function to sum and sort values of a category for specified column
# def sort_cat_by_sum(data, column_name):
#     print(data[['category', column_name]].groupby('category').sum(column_name).sort_values(column_name, ascending=False))

                       lowest_monthly_earnings
category                                      
Shows                            126961.538462
Autos & Vehicles                  68300.000000
Sports                            50063.636364
Pets & Animals                    49975.500000
Film & Animation                  45785.087174
Education                         43371.311333
Comedy                            41867.536232
News & Politics                   40192.625000
Entertainment                     38973.470539
Music                             34592.348564
People & Blogs                    34177.009167
Movies                            28400.000000
Nonprofits & Activism             24400.000000
Trailers                          22600.000000
Gaming                            16957.979574
Science & Technology              12635.411765
Howto & Style                     12069.850500
Travel & Events                    7800.000000


In [87]:
# M2: Is there a relationship between more moneys based on location?
print(data[['Country', 'highest_monthly_earnings']].groupby('Country').mean('highest_monthly_earnings').sort_values('highest_monthly_earnings', ascending=False).reset_index())

                 Country  highest_monthly_earnings
0                 Latvia              5.000000e+06
1                  Italy              2.509750e+06
2               Pakistan              2.166706e+06
3                 Turkey              2.015225e+06
4            South Korea              1.566065e+06
5   United Arab Emirates              1.442786e+06
6                  Japan              1.391342e+06
7              Argentina              9.590692e+05
8                Germany              9.399753e+05
9                  India              8.880131e+05
10              Colombia              8.734455e+05
11              Barbados              7.317000e+05
12                Russia              7.232832e+05
13                Kuwait              6.316000e+05
14         United States              5.802160e+05
15                Sweden              5.738000e+05
16           Philippines              5.674000e+05
17                Jordan              4.975000e+05
18               Vietnam       

**Help/Outreach (H)**
1) What are some channels in each category that we could potentially reach out to for help/networking?
2) What channels are on the lower ranks in the category field that we're going into? We may want to focus on lower sub-count/rank channels because they may be more likely to collab with us.

In [33]:
# H1: What are some channels in each category that we could potentially reach out to for help/networking?
# create a function to print channels titles based on category of choice:
def find_channels(data, category_type):
    filtered_df = data[data.category == category_type]
    print(filtered_df.Youtuber)
    
# print channels for top ranked category (Entertainment)
find_channels(data, 'Entertainment')

2                       MrBeast
9                 Vlad and Niki
21                       Zee TV
36                   Marshmello
46                   Get Movies
                 ...           
962              Family Fitness
963                   Zee Tamil
964              Flowers Comedy
976              SAAIHALILINTAR
978    Heidi and Zidane HZHtube
Name: Youtuber, Length: 241, dtype: object


In [54]:
# H2: What channels are on the lower ranks in the category field that we're going into?
# can use the same function as above, the output shows top and bottom 5 channels
find_channels(data, 'Entertainment')
print()

# But we are located in the US, what if we only want channels in the category of choice that are within our home country?
def find_channels_by_country(data, category_type, country):
    print(f'{category_type} Channels in {country}')
    filtered_df = data[(data.category == category_type) & (data.Country == country)]
    print(filtered_df.Youtuber)

find_channels_by_country(data, 'Entertainment', 'United States')

2                       MrBeast
9                 Vlad and Niki
21                       Zee TV
36                   Marshmello
46                   Get Movies
                 ...           
962              Family Fitness
963                   Zee Tamil
964              Flowers Comedy
976              SAAIHALILINTAR
978    Heidi and Zidane HZHtube
Name: Youtuber, Length: 241, dtype: object

Entertainment Channels in United States
2                         MrBeast
9                   Vlad and Niki
36                     Marshmello
68               Like Nastya Show
88                   TheEllenShow
                  ...            
886                   Ryan Trahan
906                  JesseAndMike
915              Top Viral Talent
937    YoungBoy Never Broke Again
949                People Vs Food
Name: Youtuber, Length: 68, dtype: object


In [84]:
find_channels_by_country(data, 'Gaming', 'United States')

Gaming Channels in United States
110                   Markiplier
121                 SSSniperWolf
146                        Dream
265                      Preston
272                        Ninja
313                      SSundee
316                       FGTeeV
414                    AS Gaming
418               Clash of Clans
467                   IShowSpeed
514                          IGN
515                       Aphmau
537                  Guava Juice
549                 CoryxKenshin
579                  Technoblade
611                  Unspeakable
640    Mobile Legends: Bang Bang
679            League of Legends
682                  PlayStation
683                   Kwebbelkop
727                       Jesser
729                 PrestonPlayz
856                 H2ODelirious
861                   theRadBrad
869                    penguinz0
973                         LOUD
Name: Youtuber, dtype: object


**Location (L)**
1) Whats the spread of categories based on location?

In [58]:
print(data.category[data.Country == 'United States'].value_counts())

category
Music                    73
Entertainment            68
People & Blogs           38
Gaming                   26
Comedy                   22
Education                17
Film & Animation         12
Howto & Style            12
Science & Technology      6
Sports                    5
News & Politics           5
Shows                     4
Pets & Animals            2
Nonprofits & Activism     1
Name: count, dtype: int64


**Trending (T)**
1) What categories have been recently trending?
    - Consider views and subscribers.

In [83]:
# T1: What categories have been recently trending?
print(data.dtypes)
# consider: video_views_for_the_last_30_days, subscribers_for_last_30_days

rank                                         int64
Youtuber                                    object
subscribers                                  int64
video views                                float64
category                                    object
Title                                       object
uploads                                      int64
Country                                     object
Abbreviation                                object
channel_type                                object
video_views_rank                           float64
country_rank                               float64
channel_type_rank                          float64
video_views_for_the_last_30_days           float64
lowest_monthly_earnings                    float64
highest_monthly_earnings                   float64
lowest_yearly_earnings                     float64
highest_yearly_earnings                    float64
subscribers_for_last_30_days               float64
created_year                   

In [85]:
# GR1: What categories have the highest average recent growth rate?
# subscribers_for_last_30_days
print(data[['category', 'subscribers_for_last_30_days']].groupby('category').mean('subscribers_for_last_30_days').sort_values('subscribers_for_last_30_days', ascending=False).reset_index())


                 category  subscribers_for_last_30_days
0        Film & Animation                 520084.714286
1                   Shows                 500000.153846
2                  Comedy                 450005.818182
3           Entertainment                 415748.503226
4        Autos & Vehicles                 400000.000000
5                  Movies                 400000.000000
6          People & Blogs                 386439.852273
7          Pets & Animals                 325001.250000
8               Education                 308823.529412
9                Trailers                 300000.000000
10                 Sports                 288888.888889
11        News & Politics                 268095.238095
12  Nonprofits & Activism                 250000.000000
13                 Gaming                 218644.728814
14   Science & Technology                 206669.466667
15                  Music                 200919.206612
16          Howto & Style                 200000

In [86]:
# video_views_for_the_last_30_days
print(data[['category', 'video_views_for_the_last_30_days']].groupby('category').mean('video_views_for_the_last_30_days').sort_values('video_views_for_the_last_30_days', ascending=False).reset_index())


                 category  video_views_for_the_last_30_days
0                   Shows                      5.501661e+08
1    Science & Technology                      4.654797e+08
2        Autos & Vehicles                      2.732165e+08
3          Pets & Animals                      2.006161e+08
4                  Sports                      2.002255e+08
5        Film & Animation                      1.930848e+08
6               Education                      1.858635e+08
7                  Comedy                      1.853106e+08
8           Entertainment                      1.796649e+08
9                   Music                      1.790282e+08
10                 Movies                      1.731005e+08
11        News & Politics                      1.607470e+08
12               Trailers                      1.562020e+08
13         People & Blogs                      1.432201e+08
14  Nonprofits & Activism                      9.759050e+07
15                 Gaming               

## Findings:
Based on our analysis of the available data of the top 995 YouTube channels, we've found that the categories featured most within the data are entertainment, music, people & blogs, gaming, and comedy. These 5 categories hold the largest amount of subsribers within that top YouTuber data. Taking that into consideration, I believe the odds of becoming a largely known YouTube channel would be best if the channel was based on one of those categories. Personally, I don't think I'm a great musician or comedian, I do, however, very much enjoy playing video games. With our data analysis and knowing what I'm good at/enjoy doing, I would most likely start an entertainment channel based on gaming.

Networking with other YouTubers could make a huge difference in how quickly a new channel grows. Getting the outreach from an already established channel and/or getting tips and tricks from them could make a big impact on the growth rate for our new channel. With 26 total gaming channels in the US that are on the top YouTuber data list we have quite the number of options. Some gaming YouTubers on the lower end ranking of our data are PrestonPlayz, H2ODelirious, theRadBrad, penguinz0, and LOUD. I think smaller channels may be more apt to collabing with other lesser known channels. If we wanted to go ahead and reach out to the 'big boys' of the YouTube gamers within the US, some top channels are Markiplier, SSSniperWolf, Dream, Preston, and Ninja.

Another thing that was looked into was recent trending data. Our chosen category of gaming is listed 14th on a list of average increase of subscribers within the last 30 days, and 16th on a list of average video views within the last 30 days. While these specific results of our data analysis isn't what we would want to necessarily hear, we know that our chosen category still sits within the overall top categories of the data. We also know that trending data has its ebbs and flows and that what is trending this month may not be trending next month.

Overall I feel confident that an entertainment channel based on gaming is a good idea. Not only is it a top YouTuber category, but it would also be something that I would thuroughly enjoy.

Some other takeaways from our data are that the average highest and lowest monthly earnings from a gaming channel; being $271,905.40 and $16957.97 respectfully. This is something we can possibly look forward to once we make it big!